src/utils.js

```diff
@@ -407,15 +408,10 @@ module.exports = (function() {
       var flickrURL = url + "?" + queryString + signature;
 
       var req = request.post(flickrURL, function(error, response, body) {
-        // format:json does not actually work, so we need to grab the photo ID from the response XML:
-        // <?xml version="1.0" encoding="utf-8" ?>\n<rsp stat="ok">\n<photoid>.........</photoid>\n</rsp>\n
-        var data;
-        if(!body) {
-          error = error || "No body found in response";
-        } else if (body.indexOf('rsp stat="ok"')>-1) {
-          data = parseInt(body.split("<photoid>")[1].split("</photoid>")[0], 10);
-        }
-        callback(error, data);
+	      if(!body) return callback("No body found in response");
+        var data = JSON.parse(parser.toJson(body));
+	      if (data.rsp.stat === "fail") return callback(data.rsp.err,null);
```

Comment:
no need for the `null` as second arg here, so we can keep it but the we should also add it to L411, or we can just leave it off here, too.

Reply:
it's better to keep it, because is easier to understand that callback have two args, especially for people who don't know very well async calls in javascript ;)

Reply:
This is not educational code, though. If we keep it here, we need to also put it in the callback above so it becomes
```js
if(!body) {
  return callback("No body found in response", null);
}
```